### ISSUE TITLE
Realm: HSA_STATUS_ERROR_OUT_OF_RESOURCES on Crusher

### ISSUE THREAD

**Issue by syamajala** [StanfordLegion/legion#1339]
After upgrading legion and gasnet I've started seeing this error on Crusher with S3D:
```
:0:rocdevice.cpp            :2652: 4238358292712 us: 48946: [tid:0x7fffd023f700] Device::callbackQueue aborting with error : HSA_STATUS_ERROR_OUT_OF_RESOURCES: The runtime failed to allocate the necessary resources. This error may also occur when the core runtime library needs to spawn threads or create internal OS-specific events. code: 0x1008
```

Going back to the old versions of legion and gasnet seem to reproduce this error now as well. I'm not sure if something has changed on the machine. I also tried an older version of rocm, but the error still reproduces. 

The stack traces don't indicate which kernel this is coming from and only show the error is happening in rocm. Are there any flags I can try?

**Comment by eddy16112** [comment]
Can you run circuit example to see if the error is still reproducible?

**Comment by syamajala** [comment]
I was able to run circuit.

**Comment by elliottslaughter** [comment]
This is S3D, right? Does it happen on simpler Regent GPU codes?

**Comment by syamajala** [comment]
Yes, this happens with S3D. 

I am able to run Regent GPU versions of daxpy and circuit.

**Comment by elliottslaughter** [comment]
Ok, so at least the basic Regent codegen isn't entirely broken.

We may need to contact OLCF support to find out more about what this error means and what can cause it. Otherwise, we'd need a minimized reproducer to make progress.

**Comment by syamajala** [comment]
OLCF said this is an intermittent issue they have been seeing on Crusher. It seems to have gone away for me without making any changes. I did a number of runs and was able to scale from 1 - 128 nodes without issue.

**Comment by syamajala** [comment]
I am reopening this so we don't forget about.

I've been talking to OLCF and did some experiments. I took the same version of S3D and did a full clean build each time with rocm versions 5.1, 5.2 and 5.3. We see this error with 5.2 and 5.3, but 5.1 works. I gave them all three builds of S3D and they are able to reproduce this as well. They have opened a bug report with AMD and are currently investigating.
